<a href="https://colab.research.google.com/github/KacperKaszuba0608/Projects/blob/main/Guided_Project__Profitable_App_Profiles_for_the_App_Store_and_Google_Play_Markets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Profitable App Profiles for the App Store and Google Play Markets

## Introduction

The aim on this project is to find the most potential genres of apps for developing and bringing the best benefits in both the App Store and Goole Play Markets. The analysis is performed in Python.

## Opening and Exploring the Data

Over here I opening the datasets with reader function from csv library, converting result to a list and then explore the datasets with function `explore_data`. As we can see below the numbers of columns are not equal that's why I explore the names of columns and choose the most important to anlysis. If you have problem with understanding colnames you can find and read their description in this links: [App Store data]("https://www.kaggle.com/datasets/lava18/google-play-store-apps") and [Google Play data]("https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps").

In [1]:
!wget https://github.com/KacperKaszuba0608/Datasets/raw/main/googleplaystore.csv -O google_wget.csv
!wget https://raw.githubusercontent.com/KacperKaszuba0608/Datasets/main/AppleStore.csv -O apple_wget.csv

--2023-01-19 18:42:25--  https://github.com/KacperKaszuba0608/Datasets/raw/main/googleplaystore.csv
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/KacperKaszuba0608/Datasets/main/googleplaystore.csv [following]
--2023-01-19 18:42:25--  https://raw.githubusercontent.com/KacperKaszuba0608/Datasets/main/googleplaystore.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1349314 (1.3M) [text/plain]
Saving to: ‘google_wget.csv’

google_wget.csv     100%[===================>]   1.29M  --.-KB/s    in 0.009s  

2023-01-19 18:42:25 (138 MB/s) - ‘google_wget.csv’ saved [1349314/1349314]

--2023-01-19 18:

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
from csv import reader
data_google = list(reader(open('google_wget.csv')))
data_ios = list(reader(open('apple_wget.csv')))

ios_header = data_ios[0]
ios_set = data_ios[1:]

android_header = data_google[0]
android_set = data_google[1:]

In [4]:
print(ios_header, '\n')
explore_data(ios_set, 0, 2, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'] 

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 7197
Number of columns: 16


As we see above in ios dataset are 7197 rows and 16 columns which the most interesting and usefull are: `track_name`,`currency`, `price`, `rating_count_tot`, `rating_count_ver` and `prime_genre`.

In [5]:
print(android_header, '\n')
explore_data(android_set, 0, 2, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


Android data has 10841 rows and 13 columns which the most intresting for my analysis are: `App`, `Category`, `Reviews`, `Installs`, `Price`, `Content Rating` and `Genres`.

## Deleting Wrong Data - Data Cleaning

In this part of project I detect missing, dulpicate or incorrect data in ios set and android set and then I correct or delete it. In the disscusion on website with dataset's documentations I find some questions and answers about missing or duplicate data. One of them describes one row in android_set has missing value and I have to remove it with `del` function. 

In [6]:
del android_set[10472]

After removing row with missing value I read about duplicate apps in android_set. That's why I wrote a function which finds a duplicate apps. My function return count of duplicate_apps and some exampls of them.

In [7]:
def check_duplicate(data_set, header = False):
    if header:
        data_set = data_set[1:]

    duplicate_apps = []
    apps = []
    
    for app in data_set:
        app_name = app[0]
        if app_name in apps:
            duplicate_apps.append(app_name)
        else:
            apps.append(app_name)

    print(f'Length of duplicate apps = {len(duplicate_apps)}.', '\n',
          f'Some examples of duplicate apps: {duplicate_apps[0:11]}')

In [8]:
check_duplicate(android_set)

Length of duplicate apps = 1181. 
 Some examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic']


Unfrotunantely I can't remove all duplicate apps, because some of them has a different value in the `Reviews` column. That's why I have to check the reviews value and then keep duplicate apps with max reviews value.

In [9]:
def remove_wrong_data(data_set, header = False):
    reviews_max = {}
    clean_data = []
    already_added_apps = []
    
    if header:
        data_set = data_set[1:]

    # First step
    for app in data_set:
        name = app[0]
        n_reviews1 = float(app[3])
        if name in reviews_max and reviews_max[name] < n_reviews1:
            reviews_max[name] = n_reviews1
        elif name not in reviews_max:
            reviews_max[name] = n_reviews1
    
    #Second step
    for row in data_set:
        name = row[0]
        n_reviews2 = float(row[3])
        
        if (reviews_max[name] == n_reviews2) and (name not in already_added_apps):
            clean_data.append(row)
            already_added_apps.append(name)
            
    return clean_data

In [10]:
android_set2 = remove_wrong_data(android_set)

explore_data(android_set2, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


From function `check_duplicate` I know the number of duplicate apps. That's why now we can check what is expected length of Google dataset.

In [11]:
print(f'Expected length: {len(android_set) - 1181}\nActual Length: {len(android_set2)}')

Expected length: 9659
Actual Length: 9659


In [12]:
def is_english_apps(data, index_of_appname):
    clean_data = []
    
    for row in data:
        name = row[index_of_appname]
        non_ascii = 0

        for letter in name:
            if ord(letter) > 127:
                non_ascii += 1
        
        if non_ascii <= 3:
            clean_data.append(row)
            
    return clean_data

In the previous function I iterate over the value of app name (Google set - index 0, App Store set - index 1) and then I check value of every letter in app name. If the value is greater than 127 It's probably non english letter. I count the non_ascii and if it's more than 3 I accept It as a non english app otherwise I append entire row to the list created at the begining. At the end of function I return the list of clean_data with english apps only.

In [13]:
android_clean = is_english_apps(android_set2, 0)
ios_clean = is_english_apps(ios_set, 1)

explore_data(android_clean, 0, 2, True)
explore_data(ios_clean, 0, 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9614
Number of columns: 13
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 6183
Number of columns: 16


## Isolating the Free Apps

Below, I isolate free apps in terms of price. My function take 3 values:
* dataset;
* index of column where price is lodge;
* name of company - in my datasets I accept two values "iso" or "android".

I iterate over rows in data, then I check what kind of company it is and finally I check price and if the value are equal 0, I append row to list created at the begining of function.

In [14]:
def isolate_free_apps(data, index_of_price, company):
    free_apps = []
    
    for app in data:
        price = app[index_of_price]
        if company == "iso":
            if price == "0.0":
                free_apps.append(app)
        elif company == "android":
            if price == "0":
                free_apps.append(app)
            
    return free_apps

In [15]:
print(f'Length of iso data: {len(isolate_free_apps(ios_clean, 4, "iso"))}')
print(f'Length of android data: {len(isolate_free_apps(android_clean, 7, "android"))}')

Length of iso data: 3222
Length of android data: 8864


In [16]:
ios_clean = isolate_free_apps(ios_clean, 4, "iso")
android_clean = isolate_free_apps(android_clean, 7, "android")

## Most Common Apps by Genre

We want to find apps in both stores because if the app is in ios store and android store it's probably very good and it have potential to developing.

In [17]:
def frequence_table(data, index_of_genres):
    freq_table = {}
    freq_table_percent = {}
    total = len(data)
    
    for app in data:
        genre = app[index_of_genres]
        
        if genre in freq_table:
            freq_table[genre] += 1
        else:
            freq_table[genre] = 1
            
    for genre in freq_table:
        percent = round((freq_table[genre] / total) * 100, 2)
        freq_table_percent[genre] = percent
            
    return freq_table_percent

In [18]:
def display_table(dataset, index):
    table = frequence_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

**`prime_genre` in App Store dataset**

In [19]:
display_table(ios_clean, -5)

Games : 58.16
Entertainment : 7.88
Photo & Video : 4.97
Education : 3.66
Social Networking : 3.29
Shopping : 2.61
Utilities : 2.51
Sports : 2.14
Music : 2.05
Health & Fitness : 2.02
Productivity : 1.74
Lifestyle : 1.58
News : 1.33
Travel : 1.24
Finance : 1.12
Weather : 0.87
Food & Drink : 0.81
Reference : 0.56
Business : 0.53
Book : 0.43
Navigation : 0.19
Medical : 0.19
Catalogs : 0.12


The most common genre in above frequency table is Games - 58.16%. The next most common is Entertainment - 7.88%. This proves that in App Store is most apps for fun than for education. The general impression have apps for entertainment: 
* Games, 
* Entertainment, 
* Photo & Video and 
* Social Networking.

What is interesting that frequence of Education app is similar to Social Networking. This proves that App Store have mostly games apps.

Inpite this frequency table I can't recommend that games are the most developed genre in whole App Store bacuase many games can have small number of users when for example social networking app like Facebook have milions user. That's why I have to check the numbers of users in next step.

**`Category` in Google Play Store dataset**

In [20]:
display_table(android_clean, 1)

FAMILY : 18.91
GAME : 9.72
TOOLS : 8.46
BUSINESS : 4.59
LIFESTYLE : 3.9
PRODUCTIVITY : 3.89
FINANCE : 3.7
MEDICAL : 3.53
SPORTS : 3.4
PERSONALIZATION : 3.32
COMMUNICATION : 3.24
HEALTH_AND_FITNESS : 3.08
PHOTOGRAPHY : 2.94
NEWS_AND_MAGAZINES : 2.8
SOCIAL : 2.66
TRAVEL_AND_LOCAL : 2.34
SHOPPING : 2.25
BOOKS_AND_REFERENCE : 2.14
DATING : 1.86
VIDEO_PLAYERS : 1.79
MAPS_AND_NAVIGATION : 1.4
FOOD_AND_DRINK : 1.24
EDUCATION : 1.16
ENTERTAINMENT : 0.96
LIBRARIES_AND_DEMO : 0.94
AUTO_AND_VEHICLES : 0.93
HOUSE_AND_HOME : 0.82
WEATHER : 0.8
EVENTS : 0.71
PARENTING : 0.65
ART_AND_DESIGN : 0.64
COMICS : 0.62
BEAUTY : 0.6


Most common category in Google Play is FAMILY (18.91%). It's better than in App Store. Google Play Store has more category and they are scattered but still there are many apps for fun.

**`Genres` in Google Play Store dataset**

In [21]:
display_table(android_clean, -4)

Tools : 8.45
Entertainment : 6.07
Education : 5.35
Business : 4.59
Productivity : 3.89
Lifestyle : 3.89
Finance : 3.7
Medical : 3.53
Sports : 3.46
Personalization : 3.32
Communication : 3.24
Action : 3.1
Health & Fitness : 3.08
Photography : 2.94
News & Magazines : 2.8
Social : 2.66
Travel & Local : 2.32
Shopping : 2.25
Books & Reference : 2.14
Simulation : 2.04
Dating : 1.86
Arcade : 1.85
Video Players & Editors : 1.77
Casual : 1.76
Maps & Navigation : 1.4
Food & Drink : 1.24
Puzzle : 1.13
Racing : 0.99
Role Playing : 0.94
Libraries & Demo : 0.94
Auto & Vehicles : 0.93
Strategy : 0.91
House & Home : 0.82
Weather : 0.8
Events : 0.71
Adventure : 0.68
Comics : 0.61
Beauty : 0.6
Art & Design : 0.6
Parenting : 0.5
Card : 0.45
Casino : 0.43
Trivia : 0.42
Educational;Education : 0.39
Board : 0.38
Educational : 0.37
Education;Education : 0.34
Word : 0.26
Casual;Pretend Play : 0.24
Music : 0.2
Racing;Action & Adventure : 0.17
Puzzle;Brain Games : 0.17
Entertainment;Music & Video : 0.17
Casual;

Frequency table of genres in Google Play is very scattered and detailed. From this table possible is statement that in Google Play the most common are practical apps (Tools, Education, Business, Productivity etc.) the exception is Entertainment genre. From both frequency tables I can't clearly state that this

## Most Popular Apps by Genre on the App Store

In [22]:
def most_popular_apps_ios(dataset, index_of_genre, index_of_user_ratings):
    freq_table = frequence_table(dataset, index_of_genre)
    list_of_ratings = {}
    
    for genre in freq_table:
        total = 0
        len_genre = 0
        for app in dataset:
            genre_app = app[index_of_genre]
            if genre_app == genre:
                number_of_user_ratings = app[index_of_user_ratings]
                total += float(number_of_user_ratings)
                len_genre += 1
    
        avg_n_ratings = round(total / len_genre, 2)
        list_of_ratings[genre] = avg_n_ratings
        #print(genre, ':', avg_n_ratings)
    return list_of_ratings

In [23]:
from collections import Counter

c = Counter(most_popular_apps_ios(ios_clean, -5, 5))
for i in c.most_common(5):
    print(i)

('Navigation', 86090.33)
('Reference', 74942.11)
('Social Networking', 71548.35)
('Music', 57326.53)
('Weather', 52279.89)


I use `Counter` function to show 5 most popular genres in App Store. Below, I print apps which genre is `Navigation`, second print is app which genre is `Reference` and third print is `Social Networking` apps.

In [24]:
def most_popular_apps_by_genre_ios(data_set, index_of_genre, index_of_appname,
                                   index_of_user_rating, name_of_genre):
    total = 0
    all_installs = 0
    
    for i in data_set:
        genre = i[index_of_genre]
        if genre == name_of_genre:
            print(f'{i[index_of_appname]} : {i[index_of_user_rating]}')
            all_installs += float(i[index_of_user_rating])
            total += 1
    print('---------------------------------------------------')    
    avg = avg = round(all_installs / total, 2)
    print(f'Average of installs = {avg}')

### Navigation

In [25]:
most_popular_apps_by_genre_ios(ios_clean, -5, 1, 5, "Navigation")

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
---------------------------------------------------
Average of installs = 86090.33


`Navigation`

Three most count of users have apps:
1. Waze - GPS Navigation, Maps & Real-time Traffic : 345046
2. Google Maps - Navigation & Transit : 154911
3. Geocaching® : 12811

### Reference

In [26]:
most_popular_apps_by_genre_ios(ios_clean, -5, 1, 5, "Reference")

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0
---------------------------------------------------
Average of installs = 74942.11


`Reference`

Five most count of users have apps:
1. Bible : 985920
2. Dictionary.com Dictionary & Thesaurus : 200047
3. Dictionary.com Dictionary & Thesaurus for iPad : 54175
4. Google Translate : 26786
5. Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418

### Social Networking

In [27]:
most_popular_apps_by_genre_ios(ios_clean, -5, 1, 5, "Social Networking")

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

`Social Networking`

Six most count of users have apps:
1. Facebook : 2974676
2. Pinterest : 1061624
3. Skype for iPhone : 373519
4. Messenger : 351466
5. Tumblr : 334293
6. WhatsApp Messenger : 287589

## Most Popular Apps by Category on Google Play



In [28]:
def most_popular_apps_android(dataset, index_of_genre, index_of_installs):
    freq_table = frequence_table(dataset, index_of_genre)
    list_of_ratings = {}

    for category in freq_table:
        total = 0
        len_category = 0
        for app in dataset:
            category_app = app[index_of_genre]
            if category_app == category:
                number_of_installs = app[index_of_installs]
                number_of_installs = number_of_installs.replace('+', '')
                number_of_installs = number_of_installs.replace(',', '')
                total += float(number_of_installs)
                len_category += 1
    
        avg_n_ratings = round(total / len_category, 2)
        list_of_ratings[category] = avg_n_ratings
        
    return list_of_ratings

In [29]:
c = Counter(most_popular_apps_android(android_clean, 1, 5))

for i in c.most_common(5):
    print(i)

('COMMUNICATION', 38456119.17)
('VIDEO_PLAYERS', 24727872.45)
('SOCIAL', 23253652.13)
('PHOTOGRAPHY', 17840110.4)
('PRODUCTIVITY', 16787331.34)


In [30]:
def most_popular_apps_by_genre_android(data_set, index_of_genre, 
                                       index_of_appname,
                                       index_of_user_rating,
                                       name_of_genre):
    total = 0
    all_installs = 0

    for app in data_set:
        genre = app[index_of_genre]
        rating = app[index_of_user_rating]
        if genre == name_of_genre and (rating == '1,000,000,000+'
                                  or rating == '500,000,000+'
                                  or rating == '100,000,000+'):
            print(f'{app[index_of_appname]} : {rating}')
            
            n_installs = rating.replace(',', '')
            n_installs = n_installs.replace('+', '')
            all_installs += float(n_installs)
            total += 1
            
    print('---------------------------------------------------')    
    avg = round(all_installs / total, 2)
    print(f'Average of installs = {avg}')

### COMMUNICATION

In [31]:
most_popular_apps_by_genre_android(android_clean, 1, 0, 5, "COMMUNICATION")

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

`COMMUNICATION`

Apps over 1,000,000,000 numbers of installs:
1. WhatsApp Messenger
2. Messenger – Text and Video Chat for Free
3. Skype - free IM & video calls
4. Google Chrome: Fast & Secure
5. Gmail
6. Hangouts

As we see above, six of giant apps outweigh the rest of apps but we can see three apps are the same like in App Store (WhatsApp Messenger, Messenger and Skype). When we remove above apps we can count real average of installs in genre `COMMUNICATION`.

In [32]:
total = 0
all_installs = 0

for app in android_clean:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        all_installs += float(n_installs)
        total += 1
                                       
avg = round(all_installs / total, 2)
print(f'Average of installs = {avg}')

Average of installs = 3603485.39


It's easy to see the change of average. Now, it's better and more subjective.

### SOCIAL

In [33]:
most_popular_apps_by_genre_android(android_clean, 1, 0, 5, "SOCIAL")

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+
---------------------------------------------------
Average of installs = 369230769.23


Results are similar too App Store. That's we can say that this apps are recomended to developing. In case of average we have to do the same thing as with results from communication genre.

In [34]:
total = 0
all_installs = 0

for app in android_clean:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'SOCIAL') and (float(n_installs) < 100000000):
        all_installs += float(n_installs)
        total += 1
                                       
avg = round(all_installs / total, 2)
print(f'Average of installs = {avg}')

Average of installs = 3084582.52


As we see average is much better than before.

### BOOK AND REFERENCE

In [35]:
total = 0
all_installs = 0

for app in android_clean:
    genre = app[1]
    rating = app[5]
    if genre == "BOOKS_AND_REFERENCE" and (rating != '1,000,000,000+'):
        print(f'{app[0]} : {rating}')
            
        n_installs = rating.replace(',', '')
        n_installs = n_installs.replace('+', '')
        all_installs += float(n_installs)
        total += 1
            
print('---------------------------------------------------')    
avg = round(all_installs / total, 2)
print(f'Average of installs = {avg}')

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
English translation from Bengali : 1

## Conclusion

To sum up in both stores the most profitable genres are:

1. Communication and Social Network;
2. Reference despite average of installs in Goole Play are much higher.